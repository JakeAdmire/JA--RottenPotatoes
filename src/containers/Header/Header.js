import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

export const Header = (props) => {

  return (
    <div className="Header">
      <div className="title">
        <h1>Rotten Potatoes</h1>
      </div>
      <div className="nav">
        <p>Showing Results for:</p>
        <p>{ props.location && props.location.name }</p>
        <NavLink to="/">change location..</NavLink>
      </div>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  location: state.location
})

export default connect(mapStateToProps, null)(Header);