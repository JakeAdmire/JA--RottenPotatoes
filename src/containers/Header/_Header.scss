.Header {
  position: fixed;
  height: 170px; width: 100%;
  color: white;
  display: flex;
  flex-direction: column;

  .title {
    height: 125px;
    background: #404040;

    h1 {
      font-family: 'Lobster', cursive;
      font-size: 80px;
      font-weight: 500;
      text-decoration: underline;
      text-shadow: 0 3px 6px rgba(0, 0, 0, 0.25);
      margin: 0 0 0 20px;
    }
  }

  .nav {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 40px;
    background: #00000080;
    box-shadow: 0 3px 6px #00000040;

    * { margin: 0 5px; };

    p:first-child {
      font-weight: bold;
    }

    button {
      border: none;
      background: none;
      color: #BFBFBF;
      font-size: 14px;

      &:hover { cursor: pointer; }
    }
  }
}

