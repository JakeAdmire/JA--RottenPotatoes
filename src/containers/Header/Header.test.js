import React from 'react';
import { shallow } from 'enzyme';

import { Header, mapStateToProps } from './Header';

describe('Header', () => {

  let wrapper;
  let mockSetRedirect = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Header setRedirect={mockSetRedirect} />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {

    let mockState = { location: '' };
    let mockProps = { location: mockState.location };
  
    it('should return a props object', () => {
      const results = mapStateToProps(mockState);
      expect(results).toEqual(mockProps);
    })
  
  })

})