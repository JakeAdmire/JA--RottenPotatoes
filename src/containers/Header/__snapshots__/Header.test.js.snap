// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    setRedirect={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          <h1>
            Rotten Potatoes
          </h1>
        </div>,
        <div
          className="nav"
        >
          <p>
            Showing Results for:
          </p>
          <p />
          <NavLink
            to="/"
          >
            change location..
          </NavLink>
        </div>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Rotten Potatoes
          </h1>,
          "className": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rotten Potatoes",
          },
          "ref": null,
          "rendered": "Rotten Potatoes",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Showing Results for:
            </p>,
            <p />,
            <NavLink
              to="/"
            >
              change location..
            </NavLink>,
          ],
          "className": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Showing Results for:",
            },
            "ref": null,
            "rendered": "Showing Results for:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "change location..",
              "to": "/",
            },
            "ref": null,
            "rendered": "change location..",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            <h1>
              Rotten Potatoes
            </h1>
          </div>,
          <div
            className="nav"
          >
            <p>
              Showing Results for:
            </p>
            <p />
            <NavLink
              to="/"
            >
              change location..
            </NavLink>
          </div>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Rotten Potatoes
            </h1>,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rotten Potatoes",
            },
            "ref": null,
            "rendered": "Rotten Potatoes",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Showing Results for:
              </p>,
              <p />,
              <NavLink
                to="/"
              >
                change location..
              </NavLink>,
            ],
            "className": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Showing Results for:",
              },
              "ref": null,
              "rendered": "Showing Results for:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "change location..",
                "to": "/",
              },
              "ref": null,
              "rendered": "change location..",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
